src/S3/PostObjectV4.php

```diff
@@ -0,0 +1,209 @@
+<?php
+namespace Aws\S3;
+
+use Aws\Credentials\CredentialsInterface;
+use GuzzleHttp\Psr7\Uri;
+use Aws\Signature\SignatureTrait;
+use Aws\Common\Enum\DateFormat;
+use Aws\Api\TimestampShape as TimestampShape;
+
+/**
+ * Encapsulates the logic for getting the data for an S3 object POST upload form
+ *
+ * @link http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html
+ * @link http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-post-example.html
+ */
+class PostObjectV4
+{
+    use SignatureTrait;
+
+    const ISO8601_BASIC = 'Ymd\THis\Z';
+
+    private $ldt;
+    private $sdt;
+    private $client;
+    private $bucket;
+    private $formAttributes;
+    private $formInputs;
+    private $jsonPolicy;
+
+    /**
+     * Constructs the PostObject.
+     *
+     * The options array accepts the following keys:
+     * @link http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html
+     *
+     * @param S3ClientInterface $client     Client used with the POST object
+     * @param string            $bucket     Bucket to use
+     * @param array             $formInputs Associative array of form input
+     *                                      fields.
+     * @param array             $options    Policy condition options
+     * @param mixed             $expiration Upload expiration time value. By
+     *                                      default: 1 hour vaild peroid.
+     */
+    public function __construct(
+        S3ClientInterface $client,
+        $bucket,
+        array $formInputs,
+        array $options = [],
+        $expiration = NULL
```

Comment:
`null` would to be lowercase. The SDK follows the [PSR-2 coding style](http://www.php-fig.org/psr/psr-2/#2-5-keywords-and-true-false-null).

But I see below that you're interpreting a null as meaning `'+1 hours'`. Why not just make that string the default value?